src/app.js

```diff
@@ -8,6 +8,235 @@ import './css/style.css';
+const reportStatus = function reportStatus(status, message) {
+  // console.log(`Reporting ${ status } status: ${ message }`);
+
+  // TODO: use an underscore method for this
+  // Should probably use an Underscore template here.
+
+  const statusHTML = `<li class="${ status }">${ message }</li>`;
+
+  // note the symetry with clearStatus()
+  $('#status-messages ul').append(statusHTML);
+  $('#status-messages').show();
+};
+
+// function for saving model
+const saveModel = function saveModel(modelToAdd, collection = 'none') {
+  modelToAdd.save({}, {
+    success: (model, response) => {
+      console.log(`Successfully saved!`);
+      if(collection != 'none') {
+        collection.add(model);
+      }
+      reportStatus('success', `Successfully saved!`);
+    },
+    error: (model, response) => {
+      console.log('Failed to save! Server response:');
+      console.log(response);
+      const errors = response.responseJSON["errors"];
+      handleValidationFailures(response.responseJSON["errors"]);
+    },
+  });
+};
+
+
+
+const render = function render(trips) {
+
+  const tripTableElement = $('#trip-list');
+  tripTableElement.html('');
+  trips.forEach((trip) => {
+    // console.log(trip)
+    const generatedHTML = tripTemplate(trip.attributes);
+
+    tripTableElement.append(generatedHTML);
+    // console.log(trip.attributes.id);
+
+    // $('#trip-list ').attr('trip-id').val(`${trip.attributes.id}`);
+    // console.log(`${trip.attributes.id}`)
+    // data-id=${trip.id}
+  });
+  $('#trips').show();
+
+  // provide visual feedback for sorting
+  $('th.sort').removeClass('current-sort-field');
+  $(`th.sort.${ trips.comparator }`).addClass('current-sort-field');
+};
+
+const showTrip = function showTrip(id) {
+  const singleTrip = $('#show-trip');
+
+  const trip = trips.findWhere({id: parseInt(id)});
+
+  trip.fetch({
+    success: (model, response) => {
+      console.log('Successfully found trip!');
+      console.log(response);
+
+      const generatedHTML = showTripTemplate(response);
+      singleTrip.html(generatedHTML);
+      $('#reserve-trip-form').on('submit', function(ev) {
+        ev.preventDefault();
+        addReservationHandler();
+      })
+    },
+  })
+  $('#show-trip').show();
+
+  console.log(`this is the show trip click event`);
+  console.log(event);
+  console.log(`this is the show trip id ${id}`);
+};
+
+// for any form where reading fields
+const readFormData = function readFormData(elementId, fields) {
+  const Data = {};
+
+  fields.forEach((field) => {
+
+    const inputElement = $(`#${elementId} input[name="${ field }"]`);
+    const value = inputElement.val();
+
+    Data[field] = value;
+    // TODO: see below
+    // clears the field
+    // break this out into a clear inputs and a method that reads inputs and one that does both
+    // methods that don't have side effects
+    // pure functions are guaranteed to be idempotent
+    //TODO: check
+    // inputElement.val('');
```

Comment:
Good that you need to recognize that you want to clear form field values, but better to wait until you have a successful save.